pkg/deploy/deploy_rp.go

```diff
@@ -90,6 +92,21 @@ func (d *deployer) DeployRP(ctx context.Context) error {
 			continue
 		}
 		if err != nil {
+			// Try to remove failed VMSS so naming conflict doesn't occur during retry
+			vmssName := "rp-vmss-" + d.version
+			innerErr := d.rpRemoveFailedScaleset(ctx, vmssName)
+			if innerErr != nil {
+				// Log instead of return so we don't mask original failure and so we can retry
+				d.log.Print(innerErr)
+			}
+			if i < 2 {
+				// Check if vmss is Deleting; if so, wait
+				vmss, innerErr := d.vmss.Get(ctx, d.config.RPResourceGroupName, vmssName)
+				if innerErr == nil && *vmss.ProvisioningState == string(mgmtcompute.ProvisioningStateDeleting) {
+					time.Sleep(10 * time.Minute)
+				}
```

Comment:
Sounds like a job for `wait.PollImmediate`

Reply:
You're right, that is the pattern we use. I'll update it.